Web.md

```diff
@@ -22,6 +22,73 @@ and retrieve compiled modules from offline storage, instantiate compiled modules
 with JavaScript imports, call the exported functions of instantiated modules,
 alias the exported memory of instantiated modules, etc.
 
+The Web embedding includes additional methods useful in that context.
+Non-web embeddings are not required to support these additional methods.
+
+### Additional Web Embedding API
+
+#### `WebAssembly.compile`
+
+In Web embeddings, the following overload is added (in addition to the core
+JS API method of the same name).
+
+```
+Promise<WebAssembly.Module> compile(Promise<Response> source)
+```
+
+The source of the material to compile is a Promise of a
+[`Response`](https://fetch.spec.whatwg.org/#response-class).
+If when unwrapped that `Promise` is not a `Response` object, then the returned `Promise` is
+[rejected](http://tc39.github.io/ecma262/#sec-rejectpromise)
+with a [`TypeError`](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror).
+
+This function starts an asynchronous task to compile a `WebAssembly.Module`
+as described in the [`WebAssembly.Module` constructor](#webassemblymodule-constructor).
+On success, the `Promise` is [fulfilled](http://tc39.github.io/ecma262/#sec-fulfillpromise)
+with the resulting `WebAssembly.Module` object. On failure, the `Promise` is
+[rejected](http://tc39.github.io/ecma262/#sec-rejectpromise) with a
+`WebAssembly.CompileError`.
+
+The `Promise<Response>` is used as the source of the bytes to compile.
+WebAssembly `source` data is expected to have a MIME type of `application/wasm`.
+Other MIME types
+[reject](http://tc39.github.io/ecma262/#sec-rejectpromise) the Promise with a
+`WebAssembly.CompileError`.
+
+#### `WebAssembly.instantiate`
+
+In Web embeddings, the following overload is added (in addition to the core
+JS API method of the same name).
+
+```
+Promise<{module:WebAssembly.Module, instance:WebAssembly.Instance}>
+  instantiate(Promise<Response> source [, importObject])
+```
+
+The source of the material to compile and instantiate is a Promise of a
```

Comment:
To support direct use of `Response` you'll need to add wrapping, e.g.: https://w3c.github.io/ServiceWorker/#fetch-event-respondwith